### PR TITLE
feat(chatbot): configure `userRoleRequired` for the `SlackChannelConfiguration`

### LINKED ISSUE #30403: chatbot: support for `UserRoleRequired` at the `SlackChannelConfiguration` class
### Describe the feature

Specify `userRoleRequired` property at the `SlackChannelConfiguration` class.

### Use Case

`SlackChannelConfiguration` class does not support for specifying `userRoleRequired` property.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.141.0

### Environment details (OS name and version, etc.)

irrelevant

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #30403.

### Reason for this change

`chatbot.SlackChannelConfiguration` does not support for configuring `userRoleRequired` prop.

### Description of changes

Add `userRoleRequired` to `SlackChannelConfigurationProps`

### Description of how you validated changes

Add both unit and integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-chatbot/README.md [hunk 1]
```diff
@@ -47,3 +47,20 @@ correct log retention period (never expire, by default).
 
 By default slack channel will use `AdministratorAccess` managed policy as guardrail policy.
 The `guardrailPolicies` property can be used to set a different set of managed policies.
+
+## User Role Requirement
+
+Administrators can [require user roles](https://docs.aws.amazon.com/chatbot/latest/adminguide/understanding-permissions.html#user-role-requirement) for all current channel members and channels and all channels created in the future by enabling a user role requirement.
+
+You can configure this feature by set `userRoleRequired` property.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-chatbot/README.md:55]:
```suggestion
You can configure this feature by setting the `userRoleRequired` property.
```

**Review by lpizzinidev — APPROVED** [review]
Looks good to me 👍 Just a minor documentation change.

**Comment by badmintoncryer** [comment]
@lpizzinidev Thank you for your review! I've reflected your suggestion.

**Review by GavinZZ — APPROVED** [review]
LGTM, thanks for supporting this prop!

**Pull request merged**